datasets/aster/aster.py

```diff
@@ -0,0 +1,243 @@
+import itertools
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+import geopandas
+import numpy
+import orjson
+import planetary_computer
+import rasterio
+import stac_geoparquet
+import stactools.aster
+import stactools.aster.utils
+from adlfs import AzureBlobFileSystem
+from pystac import Collection, Item
+from pystac.extensions.eo import EOExtension
+from pystac.extensions.projection import ProjectionExtension
+from pystac.extensions.raster import RasterBand, RasterExtension
+from stactools.aster.constants import NO_DATA
+
+from pctasks.core.models.base import PCBaseModel
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection as PcTasksCollection
+from pctasks.task.context import TaskContext
+from pctasks.task.task import Task
+
+handler = logging.StreamHandler()
+handler.setFormatter(logging.Formatter("[%(levelname)s]:%(asctime)s: %(message)s"))
+handler.setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+logger.addHandler(handler)
+logger.setLevel(logging.INFO)
+
+
+class GetPartitionsInput(PCBaseModel):
+    url: str
+    asset_key: str
+
+
+class GetPartitionsOutput(PCBaseModel):
+    assets: List[Dict[str, Any]]
+
+
+class GetPartitionsTask(Task[GetPartitionsInput, GetPartitionsOutput]):
+    _input_model = GetPartitionsInput
+    _output_model = GetPartitionsOutput
+
+    def run(
+        self, input: GetPartitionsInput, context: TaskContext
+    ) -> GetPartitionsOutput:
+        collection = Collection.from_file(input.url)
+        asset = collection.assets[input.asset_key]
+        base_path = "/".join(asset.href.split("/")[-2:])
+        asset = planetary_computer.sign_asset(asset)
+        filesystem = AzureBlobFileSystem(**asset.extra_fields["table:storage_options"])
+        assets = list()
+        for i, path in enumerate(filesystem.ls(base_path)):
+            partition_asset = asset.clone()
+            partition_asset.href = f"abfs://{path}"
+            asset_as_dict = partition_asset.to_dict()
+            asset_as_dict["partition-number"] = i
+            assets.append(asset_as_dict)
+        return GetPartitionsOutput(assets=assets)
+
+
+get_partitions_task = GetPartitionsTask()
+
+
+class CreateChunksInput(PCBaseModel):
+    asset: Dict[str, Any]
```

Comment:
Maybe expand asset into a model? IIUC, we need

1. `href: str`
2. `table:storage_options: dict[str, Any]`
3. partition-number: int?

Is there a reason to use `partition-number` instead of `partition_number`? Being consistent with `Chunk` is maybe a good thing?

Reply:
So `Asset` is literally a STAC Asset, but that's not a valid member of a `PCBaseModel`. IMO it'd be a good pattern to allow `Items` and `Assets` to be inputs/outputs of `Tasks` (STAC in, STAC out), but I didn't know how to do that quickly in pydantic so I kept it simple.

Happy to update though, I'll just have to figure out how to make Asset a pydantic model (that's probably already been done somewhere, right?).

> Is there a reason to use partition-number instead of partition_number? 

Nope, just careless `kebab-case` default bias on my part. Fixed.

Reply:
> Happy to update though, I'll just have to figure out how to make Asset a pydantic model (that's probably already been done somewhere, right?).

I'm not sure, so let's punt on it.